cxplat/inc/cxplat_memory.h

```diff
@@ -113,4 +113,46 @@ cxplat_duplicate_string(_In_z_ const char* source);
 void
 cxplat_free_string(_Frees_ptr_opt_ _In_z_ const char* string);
 
+/**
+ * @brief Allocate a lookaside list.
+ *
+ * @param[in] size Allocation size of each entry in the lookaside list.
+ * @param[in] tag Tag to use for the lookaside list.
+ * @param[in] pool_flags Flags to use for the lookaside list.
+ * @param[in] allocate Optional allocation callback.
+ * @param[in] free Optional free callback.
+ * @param[out] lookaside The allocated lookaside list.
+ * @retval CXPLAT_STATUS_SUCCESS The operation was successful.
+ * @retval CXPLAT_STATUS_NO_MEMORY Unable to allocate resources for this lookaside list.
+ */
+_Must_inspect_result_ cxplat_status_t
+cxplat_allocate_lookaside_list(
```

Comment:
I want to push back on the way this abstraction is set up. We do not need to "hide" the internals of structures here. We should define the structure (most likely platform specific, possibly a typedef) in the header, and the caller passes in preallocated memory, to an "initialize" (not "allocate") function.

Reply:
Done.